pkg/solana/txm/txm.go

```diff
@@ -303,65 +295,224 @@ func (txm *Txm) sendWithRetry(ctx context.Context, msg pendingTx) (solanaGo.Tran
-					if bump {
-						if retryStoreErr := txm.txs.AddSignature(msg.id, retrySig); retryStoreErr != nil {
-							txm.lggr.Warnw("error in adding retry transaction", "error", retryStoreErr, "id", msg.id)
-							return
-						}
-						if setErr := sigs.Set(count, retrySig); setErr != nil {
-							// this should never happen
-							txm.lggr.Errorw("INVARIANT VIOLATION", "error", setErr)
-						}
-						txm.lggr.Debugw("tx rebroadcast with bumped fee", "id", msg.id, "fee", getFee(count), "signatures", sigs.List())
-					}
+	// return signed tx, id, signature for use in simulation
+	return initTx, msg.id, sig, nil
+}
 
-					// prevent locking on waitgroup when ctx is closed
-					wait := make(chan struct{})
-					go func() {
-						defer close(wait)
-						sigs.Wait(count) // wait until bump tx has set the tx signature to compare rebroadcast signatures
-					}()
-					select {
-					case <-ctx.Done():
-						return
-					case <-wait:
-					}
+// buildTx constructs a new transaction with the specified fee and signs it.
+// It updates the compute unit price and appends the signature to the transaction.
+func (txm *Txm) buildTx(ctx context.Context, baseTx solanaGo.Transaction, key string, retryCount int) (solanaGo.Transaction, error) {
+	// Set fee
+	if computeUnitErr := fees.SetComputeUnitPrice(&baseTx, txm.getFee(retryCount)); computeUnitErr != nil {
+		return solanaGo.Transaction{}, computeUnitErr
+	}
 
-					// this should never happen (should match the signature saved to sigs)
-					if fetchedSig, fetchErr := sigs.Get(count); fetchErr != nil || retrySig != fetchedSig {
-						txm.lggr.Errorw("original signature does not match retry signature", "expectedSignatures", sigs.List(), "receivedSignature", retrySig, "error", fetchErr)
-					}
-				}(shouldBump, bumpCount, currentTx)
+	// Serialize the transaction message for signing
+	txMsg, marshalErr := baseTx.Message.MarshalBinary()
+	if marshalErr != nil {
+		return solanaGo.Transaction{}, fmt.Errorf("error in MarshalBinary: %w", marshalErr)
+	}
+
+	// Sign the transaction message using the key
+	sigBytes, signErr := txm.ks.Sign(ctx, key, txMsg)
+	if signErr != nil {
+		return solanaGo.Transaction{}, fmt.Errorf("error in Sign: %w", signErr)
+	}
+
+	// Append the signature to the transaction
+	var finalSig [64]byte
+	copy(finalSig[:], sigBytes)
+	baseTx.Signatures = append(baseTx.Signatures, finalSig)
+
+	return baseTx, nil
+}
+
+// getFee calculates the compute unit price for a transaction based on the number of retries.
+func (txm *Txm) getFee(count int) fees.ComputeUnitPrice {
+	// base compute unit price should only be calculated once
+	// prevent underlying base changing when bumping (could occur with RPC based estimation)
+	fee := fees.CalculateFee(
+		txm.cfg.ComputeUnitPriceDefault(),
+		txm.cfg.ComputeUnitPriceMax(),
+		txm.cfg.ComputeUnitPriceMin(),
+		uint(count), //nolint:gosec // reasonable number of bumps should never cause overflow
+	)
+	return fees.ComputeUnitPrice(fee)
+}
```

Comment:
I think this behavior is different than what was extracted from above. Looks like you're using the txm configs here where above we're using the msg configs. Even though msg configs are derived from the txm configs, these are still separate since txm level configs can be overridden at the msg level. Think you just need to pass in the `TxConfig` as a parameter and just use those. Then update passing `msg.cfg` into this method
```suggestion
func (txm *Txm) getFee(count int, cfg TxConfig) fees.ComputeUnitPrice {
	// base compute unit price should only be calculated once
	// prevent underlying base changing when bumping (could occur with RPC based estimation)
	fee := fees.CalculateFee(
		cfg.ComputeUnitPriceDefault(),
		cfg.ComputeUnitPriceMax(),
		cfg.ComputeUnitPriceMin(),
		uint(count), //nolint:gosec // reasonable number of bumps should never cause overflow
	)
	return fees.ComputeUnitPrice(fee)
}
```